### PR TITLE
Class improvements

### PR DESCRIPTION
**Pull request by aaronwolen**
A couple minor improvements to the class structure:
- now leverage R6's superclass method for modifying initialization methods
- Add base class for annotation arrays to DRY out array creation/ingestion

### CODE REVIEW

File: R/AnnotationArrayBase.R [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  private= list(
+
+    validate_matrix = function(x) {
+      stopifnot(
+        "Annotation data must be a matrix" = is_matrix(x),
+        "Annotation matrix must have defined dim names" = has_dimnames(x)
+      )
+    },
+
+    # @description Create an empty TileDB array suitable storing aligned
```
Comment (johnkerl) [inline R/AnnotationArrayBase.R:17]: suitable for storing

File: R/AnnotationArrayBase.R [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    # annotation/assay data sparse arrays with 1 or 2 string dimensions.
```
Comment (johnkerl) [inline R/AnnotationArrayBase.R:18]: data as sparse

File: tests/testthat/test_AssayMatrix.R [hunk 3]
```diff
@@ -24,7 +24,6 @@ test_that("AssayMatrix object can be created from a dgCMatrix", {
 test_that("matrices can be added to an AssayMatrixGroup", {
   uri <- withr::local_tempdir("assay-matrix-group")
   mat <- Seurat::GetAssayData(pbmc_small[["RNA"]], "counts")
-  fs::dir_delete(uri)
```
Comment (johnkerl) [inline tests/testthat/test_AssayMatrix.R:27]: intentional omission? just checking ...

**Review by johnkerl — APPROVED** [review]
🚢 + nits :)

File: tests/testthat/test_AssayMatrix.R (hunk 3 above)
Comment (aaronwolen) [reply tests/testthat/test_AssayMatrix.R:27]: Yeah, a leftover from interactive testing. When running the tests normally `withr::local_tempdir()` automatically cleans up after itself, so it's not necessary.

**Comment by aaronwolen** [comment]
@johnkerl I made one more change. I updated the `CommandsArray` class to also inherit from the new `AnnotationArray` base class. I ran some tests and it was able to leverage the same array creation/ingestion code thanks to the flexibility of `fromDataFrame()`. It also means I can drop `create_attr_from_value()` and clears the way to start removing some of those older classes.

**Pull request merged**